src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let transState = { ...state };
+  const cloneState = [];
+
+  for (const i in actions) {
+    const { type, extraData, keysToRemove } = actions[i];
+
+    if (type === 'addProperties') {
+      for (const key in extraData) {
+        transState = Object.assign(transState, { [key]: extraData[key] });
+      }
+      cloneState.push(Object.assign({}, transState));
+    }
+
+    if (type === 'removeProperties') {
+      for (const key in keysToRemove) {
```

Comment:
Using `for...in` to iterate over arrays is not recommended. `keysToRemove` is expected to be an array, so consider using `for...of` or a standard `for` loop.